tests/unittest_tests.bzl

```diff
@@ -10,12 +28,146 @@ def _basic_passing_test(ctx):
 
 basic_passing_test = unittest.make(_basic_passing_test)
 
-def _basic_failing_test(ctx):
-    """Unit tests for a basic library verification test that fails."""
-    env = unittest.begin(ctx)
+###################################
+####### change_setting_test #######
+###################################
+def _change_setting_test(ctx):
+    """Test to verify that an analysis test may change configuration."""
+    env = analysistest.begin(ctx)
 
-    asserts.equals(env, 1, 2)
+    dep_min_os_version = ctx.attr.target_under_test[0][_ChangeSettingInfo].min_os_version
+    asserts.equals(env, "1234.5678", dep_min_os_version)
 
-    return unittest.end(env)
+    return analysistest.end(env)
 
-basic_failing_test = unittest.make(_basic_failing_test)
+_ChangeSettingInfo = provider()
+
+def _change_setting_fake_rule(ctx):
+    return [_ChangeSettingInfo(min_os_version = ctx.fragments.cpp.minimum_os_version())]
+
+change_setting_fake_rule = rule(
+    implementation = _change_setting_fake_rule,
+    fragments = ["cpp"],
+)
+
+change_setting_test = analysistest.make(
+    _change_setting_test,
+    config_settings = {
+        "//command_line_option:minimum_os_version": "1234.5678",
+    },
+)
+
+####################################
+####### failure_testing_test #######
+####################################
+def _failure_testing_test(ctx):
```

Comment:
What are other examples of legit failures one might test for?

Reply:
Well, you can theoretically test for any sort of error (including an index out of bounds exception), but I think those would be pretty terrible practice. I think the only kind of error we want to encourage expecting is one caused by fail().